### PR TITLE
Update function doc strings

### PR DESCRIPTION
**Pull request by JackCaoG**

### CODE REVIEW

**Comment by JackCaoG** [comment]
I check the generated output and it looks reasonable.

File: docs/source/index.rst [hunk 1]
```diff
@@ -77,6 +77,7 @@ distributed
 .. automodule:: torch_xla.distributed.parallel_loader
 .. autoclass:: ParallelLoader
```
Comment (will-cromar) [inline docs/source/index.rst:78]: Let's remove `ParallelLoader` from the docs. You should almost never use it instead of `MpDeviceLoader`

**Comment by JackCaoG** [comment]
@will-cromar can you take another look at this pr?

File: torch_xla/runtime.py [hunk 2]
```diff
@@ -313,8 +324,8 @@ def initialize_cache(path: str, readonly: bool = False):
   before any computations have been performed.
 
   Args:
-    path: The path at which to store the persistent cache.
-    readonly: Whether or not this worker should have write access to the cache.
+    path (str): The path at which to store the persistent cache.
```
Comment (will-cromar) [inline torch_xla/runtime.py:327]: I don't think it's necessary to include the type in the docstring when it's already annotated. (`readonly` is an optional bool, not just a bool)

**Review by will-cromar — APPROVED** [review]

File: torch_xla/runtime.py (hunk 2 above)
Comment (JackCaoG) [reply torch_xla/runtime.py:327]: yea then the type annotation is also incorrect haha, let me fix that in a follow up pr.

**Pull request merged**